includes/class-wc-emails.php

```diff
@@ -269,6 +270,108 @@ public function order_details( $order, $sent_to_admin = false, $plain_text = fal
 	}
 
 	/**
+	 * Adds Schema.org markup for order in JSON-LD format.
+	 *
+	 * @since 2.6.0
+	 * @param mixed $order
+	 * @param bool $sent_to_admin (default: false)
+	 * @param bool $plain_text (default: false)
+	 */
+	public function order_schema_markup( $order, $sent_to_admin = false, $plain_text = false ) {
+		if ( $plain_text ) {
+			return;
+		}
+
+		$accepted_offers = array();
+
+		foreach ( $order->get_items() as $item ) {
+			if ( ! apply_filters( 'woocommerce_order_item_visible', true, $item ) ) {
+				continue;
+			}
+
+			$product = apply_filters( 'woocommerce_order_item_product', $order->get_product_from_item( $item ), $item );
+			$is_visible = $product && $product->is_visible();
+
+			$item_offered = array(
+				'@type' => 'Product',
+				'name' => apply_filters( 'woocommerce_order_item_name', $item['name'], $item, $is_visible )
+			);
+
+			if ( $sku = $product->get_sku() ) {
+				$item_offered['sku'] = $sku;
+			}
+
+			if ( $is_visible ) {
+				$item_offered['url'] = get_permalink( $product->get_id() );
+			}
+
+			if ( $image_id = $product->get_image_id() ) {
+				$item_offered['image'] = wp_get_attachment_image_url( $image_id, 'thumbnail' );
+			}
+
+			$accepted_offer = array(
+				'@type' => 'Offer',
+				'itemOffered' => $item_offered,
+				'price' => $order->get_line_subtotal( $item ),
+				'priceCurrency' => $order->get_order_currency(),
+				'eligibleQuantity' => array(
+					'@type' => 'QuantitativeValue',
+					'value' => apply_filters( 'woocommerce_email_order_item_quantity', $item['qty'], $item )
+				)
+			);
+
+			$accepted_offers[] = $accepted_offer;
+		}
+
+		$markup = array(
+			'@context' => 'http://schema.org',
+			'@type' => 'Order',
+			'merchant' => array(
+				'@type' => 'Organization',
+				'name' => get_bloginfo( 'name' )
+			),
+			'orderNumber' => strval( $order->get_order_number() ),
+			'acceptedOffer' => $accepted_offers,
+			'url' => $order->get_view_order_url()
+		);
+
+		switch ( $order->get_status() ) {
+			case 'pending':
+				$markup['orderStatus'] = 'http://schema.org/OrderPaymentDue';
+				break;
+			case 'processing':
+				$markup['orderStatus'] = 'http://schema.org/OrderProcessing';
+				break;
+			case 'on-hold':
+				$markup['orderStatus'] = 'http://schema.org/OrderProcessing';
```

Comment:
Not sure if OrderProcessing is the correct type for the 'On Hold' status. A complete list of possible order statuses can be found here: https://schema.org/OrderStatus.